framework/aster-frame/src/arch/x86/cpu.rs

```diff
@@ -95,6 +95,10 @@ impl CpuSet {
     }
 }
 
+pub struct UserPreemption {
```

Comment:
Unless there is a good reason, keep the definition of a type close to its implementation block.